irohad/main/main.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "util/filesystem.hpp"
+
+#include <gflags/gflags.h>
+#include <gflags/gflags_completions.h>
+#include "main/config/impl/gflags_config.hpp"
+#include "main/raw_block_insertion.hpp"
+#include "main/service.hpp"
+
+#include "logger/logger.hpp"
+
+using iroha::config::Config;
+using iroha::config::GFlagsConfig;
+using Iroha = Service;
+using iroha::filesystem::util::read_file;
+
+int main(int argc, char *argv[]) {
+  auto log = logger::log("MAIN");
+
+  try {
+#ifdef IROHA_VERSION
```

Comment:
this better to be hardcoded into the binary

Reply:
It is hardcoded in binary. But is setted dynamically with cmake:
```
cmake .. -DIROHA_VERSION="v0.95-build-${CI_BUILD_NUMBER}-${CI_COMMIT}"
```

useful if we want to build redistributable binaries in circle ci

Reply:
And someone may build the same source code with the different version. That shouldn't be done